### PR TITLE
Extend retrieve current NEG endpoints to also look for NEGs in non-default subnets.

### PR DESCRIPTION
**Pull request by sawsa307**
/assign @gauravkghildiyal

### CODE REVIEW

**Comment by gauravkghildiyal** [comment]
Thanks!

Haven't checked the tests, but besides those, this one is looking relatively straightforward.

File: pkg/neg/syncers/transaction.go [hunk 1]
```diff
@@ -883,6 +954,18 @@ func (s *transactionSyncer) computeEPSStaleness(endpointSlices []*discovery.Endp
 	}
 }
 
+func (s *transactionSyncer) getNonDefaultSubnetName(subnet string) (string, error) {
```
Comment (mmamczur) [inline pkg/neg/syncers/transaction.go:957]: this must be able to detect if it's running for L4 or L7 NEG and use the appropriate namer

Comment (sawsa307) [reply pkg/neg/syncers/transaction.go:957]: Ahh I see, we can probably check the NEG type and use the correct namer(namer/l4namer). Both of them are available to NEG controller. Thanks for catching this!

Comment (gauravkghildiyal) [reply pkg/neg/syncers/transaction.go:960]: Thanks for catching this!

File: pkg/neg/syncers/utils_test.go [hunk 2]
```diff
@@ -891,6 +908,145 @@ func TestRetrieveExistingZoneNetworkEndpointMap(t *testing.T) {
...
+					"foo": "bar",
+				},
+				endpoint4: labels.PodLabelMap{
+					"foo": "bar",
+				},
+			},
+			expectErr: false,
+		},
+		{
+			desc: "2 empty negs in an addtional subnet, and 2 negs in the default subnet with multiple endpoints",
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/utils_test.go:940]: I didn't understand this description compared to what the test is doing or checking. Can you maybe add a comment about what that "2" figure is supposed to represent?

**Review by gauravkghildiyal — COMMENTED** [review]
Almost ready!

File: pkg/neg/syncers/utils_test.go (hunk 2 above)
Comment (sawsa307) [reply pkg/neg/syncers/utils_test.go]: Updated the description, thank you!

**Review by gauravkghildiyal — COMMENTED** [review]
Thanks

/lgtm

**Pull request merged**